### PR TITLE
Modifications in MCT to support MOSART stratification

### PR DESCRIPTION
**Pull request by bishtgautam**
Adds new fields for supporting MOSART river stratification in E3SM

Test suite: e3sm_land_developer
Test baseline: 
Test namelist changes: 
Test status: 

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review: @jgfouca

### CODE REVIEW

**Comment by bishtgautam** [comment]
@jgfouca Do I need to run a regression test suite? If yes, could you provide instructions on how to do so?

**Comment by jgfouca** [comment]
@bishtgautam your best bet is to set your cime submodule to this branch and then run some E3SM tests that have a compset that includes rof.

**Comment by jgfouca** [comment]
Let me know when you have run some tests and I will merge.

File: src/drivers/mct/main/prep_rof_mod.F90 [hunk 1]
```diff
@@ -489,6 +492,7 @@ subroutine prep_rof_merge(l2x_r, a2x_r, fractions_r, x2r_r, cime_model)
        index_x2r_Flrl_rofsub = mct_aVect_indexRA(x2r_r,'Flrl_rofsub' )
        index_x2r_Flrl_rofdto = mct_aVect_indexRA(x2r_r,'Flrl_rofdto' )
        index_x2r_Flrl_rofi   = mct_aVect_indexRA(x2r_r,'Flrl_rofi' )
+       index_x2r_Flrl_demand = mct_aVect_indexRA(x2r_r,'Flrl_demand' )
```
Comment (billsacks) [inline src/drivers/mct/main/prep_rof_mod.F90:495]: What is the reason for this addition? It looks like `index_x2r_Flrl_demand` is already set in a conditional on e3sm a few lines up, and it looks like it's only needed if running with e3sm.

Comment (bishtgautam) [reply src/drivers/mct/main/prep_rof_mod.F90]: I have removed the duplicate line.

**Comment by bishtgautam** [comment]
@jgfouca, I will run the e3sm_land_developer test suite and report back.

**Comment by bishtgautam** [comment]
In the future, I won't do a force push on open PR.

**Comment by bishtgautam** [comment]
@jgfouca The e3sm_land_developer test ran successfully on Cori.

**Pull request merged**